packages/react-hooks/src/useAsyncInterval.ts

```diff
@@ -0,0 +1,72 @@
+import { useCallback, useEffect, useRef } from 'react';
+import Log from '@deephaven/log';
+
+const log = Log.module('useAsyncInterval');
+
+/**
+ * Calls the given async callback at a target interval.
+ *
+ * If the callback takes less time than the target interval, the target interval
+ * for the next tick will be adjusted to target the remaining time in the current
+ * interval.
+ *
+ * e.g. If the target interval is 1000ms, and the callback takes 50ms to resolve,
+ * the next tick will be scheduled for 950ms from now via `setTimeout(callback, 950)`.
+ *
+ * If the callback takes longer than the target interval, the next tick will be
+ * scheduled immediately via `setTimeout(callback, 0)`. In such cases, the time
+ * between ticks may be > than the target interval, but this guarantees that
+ * a callback won't be scheduled until after the previous one has resolved.
+ * @param callback Callback to call at the target interval
+ * @param targetIntervalMs Target interval in milliseconds to call the callback
+ */
+export function useAsyncInterval(
+  callback: () => Promise<void>,
+  targetIntervalMs: number
+) {
+  const trackingRef = useRef({ count: 0, started: Date.now() });
+  const setTimeoutRef = useRef(0);
+
+  const tick = useCallback(async () => {
+    const now = Date.now();
+    let elapsedSinceLastTick = now - trackingRef.current.started;
+
+    trackingRef.current.count += 1;
+    trackingRef.current.started = now;
+
+    log.debug(
+      `tick #${trackingRef.current.count}.`,
+      elapsedSinceLastTick,
+      'ms elapsed since last tick.'
+    );
+
+    await callback();
```

Comment:
If the component unmounts during this callback, I think the timer will continue on - it returns from this callback, and there's nothing to stop it from calling `window.setTimeout` at the end.
Should set an `isCancelled` ref when unmounted, or you could overload the `setTimeoutRef` instead.